src/state/BoardBoss.js

```diff
@@ -0,0 +1,109 @@
+import { ROWS, COLUMNS } from "../constants";
+
+export const SELECTED = "^";
+
+export const EMPTY = "";
+export const PHAROH = "P";
+export const DJED = "D";
+export const OBELISK = "O";
+export const PYRAMID = "P";
+
+export class InvalidSelectionError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "InvalidSelectionError";
+  }
+}
+
+export class InvalidMoveError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "InvalidMoveError";
+  }
+}
+
+export default class BoardBoss {
+  constructor() {
+    const buildEmptyBoard = () => {
+      let emptyBoard = [];
+
+      for (let r = 0; r < ROWS; r++) {
+        let newRow = [];
+
+        for (let c = 0; c < COLUMNS; c++) {
+          newRow.push(EMPTY);
+        }
+
+        emptyBoard.push(newRow);
+      }
+
+      return emptyBoard;
+    };
+
+    this.board = buildEmptyBoard();
+  }
+
+  writeSpace(row, column, value) {
```

Comment:
I would like to make these "private" but idk if that's possible in JS... would it be right to take these outside of the exported scope/class? Kinda like unexported helpers for this class?

Reply:
As you mentioned you can use `#` to mark fields/methods private
```suggestion
  #writeSpace(row, column, value) {
```

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/2865469/166132453-fcac4ac9-e71a-49e9-a13e-fb4602f8cdf2.png">

Reply:
I'm going back on private... the `writeBoard` function makes testing super easy right now.